livechat/reports/client.py

```diff
@@ -0,0 +1,457 @@
+
+from __future__ import annotations
+
+from abc import ABCMeta
+
+import httpx
+
+from livechat.utils.helpers import prepare_payload
+
+
+class ReportsApi:
+    ''' Main class that allows retrieval of client for specific Reports
+        API version. '''
+    @staticmethod
+    def get_client(token: str,
+                   version: str = '3.3',
+                   base_url: str = 'api.livechatinc.com',
+                   http2: bool = False) -> ReportsApiInterface:
+        ''' Returns client for specific Reports API version.
+
+            Args:
+                token (str): Full token with type (Bearer/Basic) that will be
+                             used as `Authorization` header in requests to API.
+                version (str): API's version. Defaults to `3.3`.
+                base_url (str): API's base url. Defaults to `api.livechatinc.com`.
+                http2 (bool): A boolean indicating if HTTP/2 support should be
+                              enabled. Defaults to `False`.
+
+            Returns:
+                ReportsAPI: API client object for specified version.
+
+            Raises:
+                ValueError: If the specified version does not exist.
+        '''
+        client = {
+            '3.3': ReportsAPI33(token, '3.3', base_url, http2),
+            '3.4': ReportsAPI34(token, '3.4', base_url, http2)
+        }.get(version)
+        if not client:
+            raise ValueError('Provided version does not exist.')
+        return client
+
+
+class ReportsApiInterface(metaclass=ABCMeta):
+    ''' Interface class. '''
+    def __init__(self, token: str, version: str, base_url: str,
+                 http2: bool) -> ReportsApiInterface:
+        self.api_url = f'https://{base_url}/v{version}/reports'
+        self.session = httpx.Client(http2=http2,
+                                    headers={'Authorization': token})
+
+    def modify_header(self, header: dict) -> None:
+        ''' Modifies provided header in session object.
+
+            Args:
+                header (dict): Header which needs to be modified.
+        '''
+        self.session.headers.update(header)
+
+    def remove_header(self, key: str) -> None:
+        ''' Removes provided header from session object.
+
+            Args:
+                key (str): Key which needs to be removed from the header.
+        '''
+        if key in self.session.headers:
+            del self.session.headers[key]
+
+    def get_headers(self) -> dict:
+        ''' Returns current header values in session object.
+
+            Returns:
+                dict: Response which presents current header values in session object.
+        '''
+        return dict(self.session.headers)
+
+
+class ReportsAPI33(ReportsApiInterface):
+    ''' Configuration API client in version 3.3 class. '''
+
+    # Chats
+    def agents_chatting_duration(self,
+                                 to: str = None,
+                                 from_param: str = None,
```

Comment:
This is a general remark for all methods: we cannot have those two parameters defined in such an inconsistent way. I assume that you didn't use `from` as it's Python's keyword but it should be handled differently to keep it consistent with `to`. One way could be to use approach similar to reports-api helpers from apptest that accepts `date_to` and `date_from` parameters.